pkg/asset/agent/image/ignitionbase.go

```diff
@@ -0,0 +1,303 @@
+		}
+	}
+
+	err = addStaticNetworkConfig(&config, nmStateConfigs.StaticNetworkConfig)
+	if err != nil {
+		return err
+	}
+
+	// Enable pre-network-manager-config.service only when there are network configs defined
+	if len(nmStateConfigs.StaticNetworkConfig) != 0 {
+		agentEnabledServices = append(agentEnabledServices, "pre-network-manager-config.service")
+	}
+
+	filesToInclude := [...]asset.File{
+		*infraEnvAsset.File,
+		*clusterImageSetAsset.File,
+		*pullSecretAsset.File,
+	}
+
+	for _, file := range filesToInclude {
+		manifestFile := ignition.FileFromBytes(filepath.Join(manifestPath, filepath.Base(file.Filename)),
+			"root", 0600, file.Data)
+		config.Storage.Files = append(config.Storage.Files, manifestFile)
+	}
+
+	err = bootstrap.AddSystemdUnits(&config, "agent/systemd/units", agentTemplateData, agentEnabledServices)
+	if err != nil {
+		return err
+	}
+
+	addMirrorData(&config, registriesConfig, registryCABundle)
+
+	a.Config = &config
+
+	if err := a.generateFile(unconfiguredIgnitionFilename); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// CopyIgnitionConfig returns a copy of the Ignition config
+// The copy is made by marshaling and unmarshaling the ignition yaml.
+func (a *IgnitionBase) CopyIgnitionConfig() (igntypes.Config, error) {
+	copyData := &igntypes.Config{}
+	data, err := yaml.Marshal(a.Config)
+	if err != nil {
+		return *copyData, errors.Wrap(err, "copy failed, failed to Marshal IgnitionBase config")
+	}
+
+	if err := yaml.UnmarshalStrict(data, copyData); err != nil {
+		return *copyData, errors.Wrap(err, "copy failed, failed to Unmarshal IgnitionBase config")
+	}
+
+	return *copyData, nil
+}
+
+// PersistToFile writes the unconfigured ignition in the assets folder.
+func (a *IgnitionBase) PersistToFile(directory string) error {
+	if a.File == nil {
+		return errors.New("attempting to persist a IgnitionBase that has not been generated")
+	}
+	unconfiguredIgnFile := filepath.Join(directory, a.File.Filename)
+
+	err := os.WriteFile(unconfiguredIgnFile, a.File.Data, 0o644) //nolint:gosec // no sensitive info
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (a *IgnitionBase) generateFile(filename string) error {
+	data, err := ignition.Marshal(a.Config)
+	if err != nil {
+		return errors.Wrap(err, "failed to Marshal Ignition config")
+	}
+	a.File = &asset.File{
+		Filename: filename,
+		Data:     data,
+	}
+	return nil
+}
+
+// Load returns the IgnitionBase from disk.
+func (a *IgnitionBase) Load(f asset.FileFetcher) (bool, error) {
+	// The IgnitionBase will not be needed by another asset so load is noop.
+	// This is implemented because it is required by WritableAsset
+	return false, nil
+}
+
+// Files returns the files generated by the asset.
+func (a *IgnitionBase) Files() []*asset.File {
+	// Return empty array because File will never be loaded.
+	return []*asset.File{}
+}
+
+func getOSImagesInfo(cpuArch string) (*models.OsImage, error) {
+	st, err := rhcos.FetchCoreOSBuild(context.Background())
```

Comment:
Recently I opened a PR to fix the lack of support for OPENSHIFT_INSTALL_OS_IMAGE_OVERRIDE (see https://github.com/openshift/installer/pull/721). Wondering if also this point should honor it?

Reply:
This function was moved from ignition.go. But yeah, we'll need to fix it. Probably not under the context of this PR.